### PR TITLE
Ignore index.php files in Doctrine entities scanning

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.6.x
| Description?  | Change annotation driver settings so that it ignores index.php files present in Entities folder (the file contains an exit which stopped all doctrine commands execution)
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | ~
| How to test?  | Install productcomments module, try to run `doctrine:mapping:info` or `doctrine:schema:update` command without this PR nothing happens, with the PR it works correctly and the module's entities are listed

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14460)

### CODE REVIEW

File: src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php [hunk 1]
```diff
@@ -85,14 +87,23 @@ private function getCompilerPassList(array $activeModules)
                 }
                 $modulePrefix = 'Module' . Inflector::camelize($moduleFolder->getFilename());
                 $moduleEntityDirectory = realpath($moduleFolder . '/src/Entity');
-                $mappingPass = DoctrineOrmMappingsPass::createAnnotationMappingDriver([$moduleNamespace], [$moduleEntityDirectory], [], false, [$modulePrefix => $moduleNamespace]);
+                $mappingPass = $this->createAnnotationMappinDriver($moduleNamespace, $moduleEntityDirectory, $modulePrefix);
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php]:
```suggestion
                $mappingPass = $this->createAnnotationMappingDriver($moduleNamespace, $moduleEntityDirectory, $modulePrefix);
```

File: src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php [hunk 2]
```diff
@@ -85,14 +87,23 @@ private function getCompilerPassList(array $activeModules)
...
                 $mappingsPassList[$moduleEntityDirectory] = $mappingPass;
             }
         }
 
         return $mappingsPassList;
     }
 
+    private function createAnnotationMappinDriver($moduleNamespace, $moduleEntityDirectory, $modulePrefix)
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php]:
```suggestion
    private function createAnnotationMappingDriver($moduleNamespace, $moduleEntityDirectory, $modulePrefix)
```

File: src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php (hunk 1 above)
Comment (jolelievre) [reply src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php]: Nice catch!

File: src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php (hunk 2 above)
Comment (jolelievre) [reply src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php]: Done!

File: src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php [hunk 3]
```diff
@@ -85,14 +87,30 @@ private function getCompilerPassList(array $activeModules)
...
+     * @param string $moduleEntityDirectory
+     * @param string $modulePrefix
+     *
+     * @return DoctrineOrmMappingsPass
+     */
+    private function createAnnotationMappingDriver($moduleNamespace, $moduleEntityDirectory, $modulePrefix)
+    {
+        $reader = new Reference('annotation_reader');
+        $driverDefinition = new Definition('Doctrine\ORM\Mapping\Driver\AnnotationDriver', [$reader, [$moduleEntityDirectory]]);
+        $driverDefinition->addMethodCall('addExcludePaths', [[$moduleEntityDirectory . '/index.php']]);
```
Comment (Quetzacoalt91) [inline src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php]: Are you sure about the double array?

**Review by Quetzacoalt91 — DISMISSED** [review]

File: src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php (hunk 3 above)
Comment (eternoendless) [inline src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php:114]: Consider adding a comment explaining that this whole method exists for the sole purpose of avoiding `exit` found in index files.

Comment (jolelievre) [reply src/PrestaShopBundle/DependencyInjection/Compiler/ModulesDoctrineCompilerPass.php:114]: You're right! Done

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @jolelievre

**Pull request merged**